spyder/plugins/updatemanager/workers.py

```diff
@@ -150,12 +151,18 @@ def start(self):
         else:
             url = pypi_url
 
+        headers = {}
+        token = os.getenv('GITHUB_TOKEN')
+        if url == github_url and token:
+            headers.update(Authorization=f"Bearer {token}")
+
         logger.info(f"Checking for updates from {url}")
         try:
-            page = requests.get(url)
+            page = requests.get(url, headers=headers)
+            _rate_limits(page)
```

Comment:
Is this really necessary for production code? Or only for testing?

Reply:
The motivation was for investigating the issue in our test suite. However, when our installers become mainstream, it may be prudent to have a debug log message in production in case users encounter a rate limit error. In production, `headers={}`, then `requests.get(url)` and  `requests.get(url, headers={})` are equivalent.